lib/matplotlib/axes/_base.py

```diff
@@ -1635,16 +1635,28 @@ def relim(self, visible_only=False):
         # Collections are deliberately not supported (yet); see
         # the TODO note in artists.py.
         self.dataLim.ignore(True)
-        self.dataLim.set_points(mtransforms.Bbox.null().get_points())
-        self.ignore_existing_data_limits = True
+        # check if there is a need to create a null bbox
+        if (self.conditionCheck()):
```

Comment:
![4ab0e605-c1f9-4761-a964-b452e28e361a](https://cloud.githubusercontent.com/assets/8809506/6322681/b1ed9696-bae7-11e4-86bd-228bf431918a.png)
good point, but I don't know why my debugger pass that condition, it's wired, so I still did not remove the function, and it might be easier to understand and debug.